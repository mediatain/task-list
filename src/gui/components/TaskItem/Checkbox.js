import React from 'react';
import styled from 'styled-components';
import { disabledColor } from '../../colors';

const StyledCheckbox = styled.div`
  display: flex;
  align-items: center;
  padding-right: 1.7rem;
  padding-left: .1rem;
  cursor: pointer;
`;

const Box = styled.div`
  position: relative;
  border: .1rem solid;
  border-radius: .25rem;
  height: 1.5rem;
  width: 1.5rem;
  color: ${disabledColor};
`;

const Mark = styled.div`
  position: absolute;
  left: .3rem;
  top: .5rem;
  width: .2rem;
  height: 1rem;
  background: ${disabledColor};
  transform: rotate(315deg);
  display: ${({ checked }) => checked ? 'block' : 'none'};

  &::after {
    position: absolute;
    left: 100%;
    bottom: 0;
    content: "";
    background: inherit;
    width: 2rem;
    height: .2rem;
  }
`;

export default function Checkbox({
  checked = false,
  onClick,
}) {
  return (
    <StyledCheckbox onClick={onClick}>
      <Box>
        <Mark checked={checked} />
      </Box>
    </StyledCheckbox>
  );
}

