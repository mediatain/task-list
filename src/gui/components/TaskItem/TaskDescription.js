import styled from 'styled-components';

export default styled.span`
  font-size: 1.5rem;
  padding: 1rem;
`;
