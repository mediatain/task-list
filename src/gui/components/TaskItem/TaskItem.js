import React from 'react';
import CheckBox from './Checkbox';
import TaskDescription from './TaskDescription';
import styled, { css } from 'styled-components';
import { disabledColor, textColorDark } from '../../colors';

const StyledTaskItem = styled.li`
  display: flex;
  flex-direction: row;
  list-style: none;
  color: ${textColorDark};
  
  ${({ done }) => done && css`
    text-decoration: line-through;
    color: ${disabledColor};
  `}
`;

export default function TaskItem({
  task,
  onCheck,
}) {
  return (
    <StyledTaskItem done={task.done}>
      <CheckBox checked={task.done} onClick={onCheck} />
      <TaskDescription>{task.description}</TaskDescription>
    </StyledTaskItem>
  );
}

