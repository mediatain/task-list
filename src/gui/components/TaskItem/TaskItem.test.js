import React from 'react';
import 'jest-styled-components';
import { render } from '@testing-library/react';
import { shallow } from 'enzyme';
import TaskItem from './TaskItem';
import Checkbox from './Checkbox';

describe('TaskItem', () => {
  const tasksMocks = [
    { id: '1', done: true, description: 'My first task' },
    { id: '2', done: false, description: 'My second task' },
    { id: '3', done: true, description: 'Aaand another task' },
  ];

  it.each(tasksMocks)('should match the snapshot for task %p', (task) => {
    const { container } = render(<TaskItem task={task} />);

    expect(container).toMatchSnapshot();
  });

  it.each(tasksMocks)('should render the task.description', (task) => {
    const taskItem = shallow(<TaskItem task={task} />);

    expect(taskItem.text()).toContain(task.description);
  });

  it.each([
    ['no checkmark', false],
    ['no checkmark', undefined],
    ['a checkmark', true],
  ])('should display %s when task done is %s', (display, done) => {
    const task = { id: '1', done, description: 'My first task' };

    const taskItem = shallow(<TaskItem task={task} />);

    const checkBox = taskItem.find(Checkbox).dive();
    const checkedMarkExists = checkBox.exists({'checked': true });
    expect(checkedMarkExists).toBe(display === 'a checkmark');
  });
});

