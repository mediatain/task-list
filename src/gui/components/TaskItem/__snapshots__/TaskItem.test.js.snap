// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskItem should match the snapshot for task {"description": "Aaand another task", "done": true, "id": "3"} 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-right: 1.7rem;
  padding-left: .1rem;
  cursor: pointer;
}

.c2 {
  position: relative;
  border: .1rem solid;
  border-radius: .25rem;
  height: 1.5rem;
  width: 1.5rem;
  color: #a4a5a4;
}

.c3 {
  position: absolute;
  left: .3rem;
  top: .5rem;
  width: .2rem;
  height: 1rem;
  background: #a4a5a4;
  -webkit-transform: rotate(315deg);
  -ms-transform: rotate(315deg);
  transform: rotate(315deg);
  display: block;
}

.c3::after {
  position: absolute;
  left: 100%;
  bottom: 0;
  content: "";
  background: inherit;
  width: 2rem;
  height: .2rem;
}

.c4 {
  font-size: 1.5rem;
  padding: 1rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  list-style: none;
  color: #000000;
  -webkit-text-decoration: line-through;
  text-decoration: line-through;
  color: #a4a5a4;
}

<div>
  <li
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <div
          class="c3"
        />
      </div>
    </div>
    <span
      class="c4"
    >
      Aaand another task
    </span>
  </li>
</div>
`;

exports[`TaskItem should match the snapshot for task {"description": "My first task", "done": true, "id": "1"} 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-right: 1.7rem;
  padding-left: .1rem;
  cursor: pointer;
}

.c2 {
  position: relative;
  border: .1rem solid;
  border-radius: .25rem;
  height: 1.5rem;
  width: 1.5rem;
  color: #a4a5a4;
}

.c3 {
  position: absolute;
  left: .3rem;
  top: .5rem;
  width: .2rem;
  height: 1rem;
  background: #a4a5a4;
  -webkit-transform: rotate(315deg);
  -ms-transform: rotate(315deg);
  transform: rotate(315deg);
  display: block;
}

.c3::after {
  position: absolute;
  left: 100%;
  bottom: 0;
  content: "";
  background: inherit;
  width: 2rem;
  height: .2rem;
}

.c4 {
  font-size: 1.5rem;
  padding: 1rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  list-style: none;
  color: #000000;
  -webkit-text-decoration: line-through;
  text-decoration: line-through;
  color: #a4a5a4;
}

<div>
  <li
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <div
          class="c3"
        />
      </div>
    </div>
    <span
      class="c4"
    >
      My first task
    </span>
  </li>
</div>
`;

exports[`TaskItem should match the snapshot for task {"description": "My second task", "done": false, "id": "2"} 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-right: 1.7rem;
  padding-left: .1rem;
  cursor: pointer;
}

.c2 {
  position: relative;
  border: .1rem solid;
  border-radius: .25rem;
  height: 1.5rem;
  width: 1.5rem;
  color: #a4a5a4;
}

.c3 {
  position: absolute;
  left: .3rem;
  top: .5rem;
  width: .2rem;
  height: 1rem;
  background: #a4a5a4;
  -webkit-transform: rotate(315deg);
  -ms-transform: rotate(315deg);
  transform: rotate(315deg);
  display: none;
}

.c3::after {
  position: absolute;
  left: 100%;
  bottom: 0;
  content: "";
  background: inherit;
  width: 2rem;
  height: .2rem;
}

.c4 {
  font-size: 1.5rem;
  padding: 1rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  list-style: none;
  color: #000000;
}

<div>
  <li
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <div
          class="c3"
        />
      </div>
    </div>
    <span
      class="c4"
    >
      My second task
    </span>
  </li>
</div>
`;
